library/system/src/modules/Kernel.rb

```diff
@@ -248,6 +253,10 @@ def ParseInstallationKernelCmdline
       @cmdline_parsed = true
       return if !(Stage.initial || Stage.cont)
       tmp = Convert.to_string(SCR.Read(path(".etc.install_inf.Cmdline")))
+      # live installation does not create /etc/install.inf (bsc#793065)
+      if Mode.live_installation
+        tmp = Convert.to_string(SCR.Read(path(".target.string"), "/proc/cmdline"))
```

Comment:
We have an specific ```.proc.cmdline``` agent. Shouldn't we use it instead of ```target.string```? I even have a forever-pending bug about it https://bugzilla.suse.com/show_bug.cgi?id=889497

On the other hand, I think that you should use if-else
```ruby
if Mode.live_installation
  # read from /proc
else
  # read from install.inf
end
```
And last but now least :-) I'm not sure if calling ```to_string``` is actually needed.